src/java/org/apache/cassandra/index/sai/disk/v1/WidePrimaryKeyMap.java

```diff
@@ -166,10 +177,7 @@ private Clustering<?> readClusteringKey(long sstableRowId)
     // Returns the rowId of the next partition or the number of rows if supplied rowId is in the last partition
     private long startOfNextPartition(long rowId)
     {
-        long partitionId = partitionArray.get(rowId);
-        long nextPartitionRowId = partitionArray.indexOf(++partitionId);
-        if (nextPartitionRowId == -1)
-            nextPartitionRowId = partitionArray.length();
-        return nextPartitionRowId;
+        long partitionSize = partitionSizeArray.get(partitionArray.get(rowId));
```

Comment:
nit: Would consider being clearer here that this is a number of rows for people skimming.

This is nice though. The idea is to avoid the binary search and just lookup the size directly?

Reply:
As above, I have changed all the variable names in these classes to make it clearer who is doing what.

As an aside, I found that locally `MonotonicBlockPackedReader.indexOf` is marked as unsupported. I think we missed that when the `indexOf` was put back in. This was causing the final issue we were having, which is why this component was added. I have also now marked that method as unsupported.